### ISSUE TITLE
PhET-iO design review

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/calculus-grapher#224]
Calculus Grapher is ready for PhET-iO design review.  

The "Calculus Grapher PhET-iO Design" doc is https://docs.google.com/document/d/1M7th-Lanv9tS7O7SPOcpR2cn10nu0IbciQqex25Go_Q/edit

In addition to an overall review, please provide feedback on these specific features:

- [x] Review elements for the "Vertical Lines" feature. Explore subtrees `model.tools.verticalLines` and `view.graphsNode.verticalLinesNode` for any screen. To run the sim with all vertical lines visible, use `?verticalLinesVisible=true`.

- [x] Review elements for the "Labeled Points" feature. Explore subtrees `model.tools.labeledPoints` and `view.graphsNode.originalGraphNode.labeledPointsNode` for any screen. To run the sim with all labeled points visible, use `?labeledPointsVisible=true`.

- [x] Screens that switch between 2 sets of graphs (via radio buttons) have `model.graphSetProperty`. See `advancedScreen.model.graphSetProperty` and `labScreen.model.graphSetProperty`.  Verify that the documentation is sufficient for `graphSetProperty` and its valid values (`graphSet0`, `graphSet1`).

- [x] Inspect the Simulation tab of the Preferences dialog, under `preferencesDialog.preferencesPanels.simulationPreferencesPanel`.

**Comment by amanda-phet** [comment]
Update: KP and I will spend an hour on our own to review, then come together to discuss questions and things that came up. We should also ask Amy to take a look.

1. Look at the structure of the tree
2. Names of things in the tree
3. Anything not instrumented that should be
4. All of the data we’d want to get from the sim in the tree/available
5. Set/Get value things in the tree
6. Anything read-only or read/write that should be changed?
7. Review featured items? Not sure what the process is yet for featuring/un-featuring. Might still do overrides file.

We will take notes in the [PhET-iO design doc](https://docs.google.com/document/d/1M7th-Lanv9tS7O7SPOcpR2cn10nu0IbciQqex25Go_Q/edit#heading=h.6t3nr69j3v8t) and reconnect to see if there were any issues that came up. We can use this issue to list specifics. 

Action items: 
@arouinfar can you review the studio tree for consistency with current PhET-iO design patterns and practices? 
@kathy-phet will review this.
@amanda-phet and @catherinecarter will pair on reviewing this.

**Comment by pixelzoom** [comment]
In case you'd like to review a dev version:
https://phet-dev.colorado.edu/html/calculus-grapher/1.0.0-dev.14/phet-io/

**Comment by amanda-phet** [comment]
I meant to ask today if this issue should be closed, since @pixelzoom opened new issues for the items we discussed today. I believe the review is complete, but @pixelzoom can you confirm we did everything you would expect? 

Side question- when/how should I feature things?

**Comment by pixelzoom** [comment]
> I meant to ask today if this issue should be closed, since @pixelzoom opened new issues for the items we discussed today. I believe the review is complete, but @pixelzoom can you confirm we did everything you would expect?

This question feels backwards to me. You're the designer. Did you verify that what we implemented matches what you designed?

If you're asking if you completed the review process...  That's difficult for me to answer. Is there a review process? After instrumenting 12+ sims, I still don't know.  

If your review consisted solely of looking at the Studio tree, then no, you're not done.  A review also includes:
* Save/load configurations of the sim in Studio, to verify that it behaves as desired. 
* Verify that all of the sim meets all of your requirements that you listed in the first sections of the [PhET-iO Design](https://docs.google.com/document/d/1M7th-Lanv9tS7O7SPOcpR2cn10nu0IbciQqex25Go_Q/edit) document.
* Complete the [Post Instrumentation Review and Checks](https://github.com/phetsims/phet-io/blob/master/doc/phet-io-instrumentation-process-checklist.md#post-instrumentation-review-and-checks) portion of the PhET-iO Instrumentation Process Checklist.

You may have done these things, but I don't see evidence in the PhET-iO Design document, or in GitHub issues.

> Side question- when/how should I feature things?

Whenever you want to, see https://github.com/phetsims/calculus-grapher/issues/225

**Comment by amanda-phet** [comment]
Additional things to do: 

- [ ] Save/load configurations of the sim in Studio, to verify that it behaves as desired.
- [ ] Verify that the sim meets the requirements that you listed in the first sections of the [PhET-iO Design](https://docs.google.com/document/d/1M7th-Lanv9tS7O7SPOcpR2cn10nu0IbciQqex25Go_Q/edit) document.
- [ ] Complete the [Post Instrumentation Review and Checks](https://github.com/phetsims/phet-io/blob/master/doc/phet-io-instrumentation-process-checklist.md#post-instrumentation-review-and-checks) portion of the PhET-iO Instrumentation Process Checklist. [edit: will mostly do these items but not all are totally relevant] 

Once these are checked off, I will close this issue.

**Comment by veillette** [comment]
FYI: I used the PhET wrapper state in PhET-IO where one interacts with a sim and observe state changes in a copy of the sim.
https://phet-dev.colorado.edu/html/calculus-grapher/1.0.0-dev.17/phet-io/wrappers/state/

I checked that all the interactive elements of the control panel, the checkboxes as well as the preferences dialog are passed on properly onto the copy of the sim. Everything here is working as it should.
